src/hwas/hwas.c

```diff
@@ -165,6 +221,7 @@ hwas_PI_InterruptManagement_DisableInterrupt_Pi(const asn1SccInterruptNumber* IN
 void
 hwas_PI_InterruptManagement_EnableInterrupt_Pi(const asn1SccInterruptNumber* IN_interrupt)
 {
+    Nvic_setInterruptPriority((Nvic_Irq)*IN_interrupt, PERIPH_INTERRUPT_PRIORITY);
```

Comment:
I am not really sure what does it do. PERIPH INTERRUPT PRIORITY is 2, HWAS_INTERRUPT_PRIORITY is 1... but can a task pre-empt an interrupt?

Reply:
That was my attempt in doing that. But it can't, actually. Those priorities are different things. The second one is used by freeRTOS to pull the task from the task list according to its priority. The first one is NVIC priority. It can be a number from 1 to 7. It cannot be 0, as for some reason, it might interfere with FreeRTOS, and assert will be called. 
So task cannot pre-empt an interrupt. Those numbers mean different things. "2" is just an arbitrary number. It can be "1" for example.

Reply:
If it cannot be left as default, please change it to 1